scripts/edit-task.js

```diff
@@ -0,0 +1,231 @@
+import { CLICK_EVENT, HIDE_CLASS } from "/scripts/constants.js";
+import createButton from "/scripts/factory/createButton.js";
+import toggleButtons from "/scripts/factory/toggleButton.js";
+
+function saveEvent(
+  doneButton,
+  editButton,
+  deleteButton,
+  saveAndDoneButton,
+  revertButton,
+  saveButton,
+  inputElement,
+  tasks,
+  id,
+  saveEventHandler,
+  saveAndDoneEventHandler,
+  revertEventHandler
```

Comment:
function takes too many parameters, consider refactoring and introducing multiple functions which does specific job rather doing everything within one function.

// see https://eslint.org/docs/latest/rules/max-params